File: VB/GrowthWare.Framework/Model/Profiles/MMessageProfile.vb
Imports GrowthWare.Framework.Model.Profiles.Base
Imports GrowthWare.Framework.Model.Profiles.Interfaces
Imports System.Reflection

Namespace Model.Profiles
    <Serializable(), CLSCompliant(True)> _
    Public Class MMessageProfile
        Inherits MProfile
        Implements IMessageProfile

#Region "Member Properties"
        Private m_SecurityEntity_Seq_Id As Integer = 1
        Private m_Description As String = String.Empty
        'Private m_ErrorCode As Integer = -1
        Private m_Title As String = String.Empty
        Private m_FormatAsHTML As Boolean = False
        Private m_Body As String = String.Empty
#End Region

#Region "Protected Methods"
        ''' <summary>
        ''' Initializes the specified data row.
        ''' </summary>
        ''' <param name="dataRow">The data row.</param>
        Private Shadows Sub Initialize(ByVal dataRow As DataRow)
            MyBase.IdColumnName = "MESSAGE_SEQ_ID"
            MyBase.NameColumnName = "Name"
            MyBase.Initialize(dataRow)
            m_SecurityEntity_Seq_Id = MyBase.GetInt(dataRow, "SE_SEQ_ID")
            m_Title = MyBase.GetString(dataRow, "TITLE")
            m_Description = MyBase.GetString(dataRow, "DESCRIPTION")
            m_FormatAsHTML = MyBase.GetBool(dataRow, "FORMAT_AS_HTML")
            m_Body = MyBase.GetString(dataRow, "BODY")
        End Sub
#End Region

#Region "Public Methods"
        ''' <summary>
        ''' Will return a message profile with the default vaules
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Initializes a new instance of the <see cref="MMessageProfile" /> class.
        ''' </summary>
        ''' <param name="dataRow">The data row.</param>
        Public Sub New(ByVal dataRow As DataRow)
            initialize(dataRow)
        End Sub

        ''' <summary>
        ''' Initializes a new instance of the <see cref="MMessageProfile" /> class.
        ''' </summary>
        ''' <param name="profile">MMessageProfile</param>
        Public Sub New(ByVal profile As MMessageProfile)
            If profile IsNot Nothing Then
                With Me
                    .AddedBy = profile.AddedBy
                    .AddedDate = profile.AddedDate
                    .Body = profile.Body
                    .Description = profile.Description
                    .FormatAsHtml = profile.FormatAsHtml
                    .Id = profile.Id
                    .Name = profile.Name
                    .SecurityEntitySeqId = profile.SecurityEntitySeqId
                    .Title = profile.Title
                    .UpdatedBy = profile.UpdatedBy
                    .UpdatedDate = profile.UpdatedDate
                End With
            End If
        End Sub

        ''' <summary>
        ''' Formats the body and replaces < and > with the appropriate property value.
        ''' </summary>
        Public Overridable Sub FormatBody() Implements IMessageProfile.FormatBody
            Dim myPropertyInfo As PropertyInfo() = Me.GetType.GetProperties()
            Dim myPropertyItem As PropertyInfo
            For Each myPropertyItem In myPropertyInfo
                Dim pValue As Object = myPropertyItem.GetValue(Me, Nothing)
                If pValue IsNot Nothing Then
                    m_Body = m_Body.Replace("<" & myPropertyItem.Name & ">", pValue.ToString)
                Else
                    m_Body = m_Body.Replace("<" & myPropertyItem.Name & ">", "")
                End If
            Next
        End Sub

        ''' <summary>
        ''' Returns all properties encapsulated by angle brackets seporated by the Seporator parameter
        ''' </summary>
        ''' <param name="separator">string</param>
        ''' <returns>string</returns>
        Public Function GetTags(ByVal separator As String) As String Implements IMessageProfile.GetTags
            Dim mRetVal As String = String.Empty
            Dim mPropertyInfo As PropertyInfo() = Me.GetType.GetProperties()
            For Each mPropertyItem As PropertyInfo In mPropertyInfo
                mRetVal = mRetVal & "<" & mPropertyItem.Name & ">" & separator
            Next
            Return mRetVal
        End Function
#End Region

#Region "Public Properties"
        ''' <summary>
        ''' Sets or gets the SecurityEntitySeqId property
        ''' </summary>
        ''' <value>Sets the value</value>
        ''' <returns>String</returns>
        ''' <remarks></remarks>
        Public Property SecurityEntitySeqId() As Integer
            Get
                Return m_SecurityEntity_Seq_Id
            End Get
            Set(ByVal value As Integer)
                m_SecurityEntity_Seq_Id = value
            End Set
        End Property

        ''' <summary>
        ''' Sets or gets the Title property
        ''' </summary>
        ''' <value>Sets the value</value>
        ''' <returns>String</returns>
        ''' <remarks></remarks>
        Public Property Title() As String Implements IMessageProfile.Title
            Get
                Return m_Title
            End Get
            Set(ByVal value As String)
                If Not value Is Nothing Then m_Title = value.Trim
            End Set
        End Property

        ''' <summary>
        ''' Sets or gets the Description property
        ''' </summary>
        ''' <value>Sets the value</value>
        ''' <returns>String</returns>
        ''' <remarks></remarks>
        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(ByVal value As String)
                If Not value Is Nothing Then m_Description = value.Trim()
            End Set
        End Property

        ''' <summary>
        ''' Sets or gets the FormatAsHtml property
        ''' </summary>
        ''' <value>Sets the value</value>
        ''' <returns>String</returns>
        ''' <remarks></remarks>
        Public Property FormatAsHtml() As Boolean Implements IMessageProfile.FormatAsHtml
            Get
                Return m_FormatAsHTML
            End Get
            Set(ByVal value As Boolean)
                m_FormatAsHTML = value
            End Set
        End Property

        ''' <summary>
        ''' Sets or gets the body property
        ''' </summary>
        ''' <value>Sets the value</value>
        ''' <returns>String</returns>
        ''' <remarks></remarks>
        Public Property Body() As String Implements IMessageProfile.Body
            Get
                Return m_Body
            End Get
            Set(ByVal value As String)
                m_Body = value
            End Set
        End Property
#End Region
    End Class
End Namespace

File: CS/GrowthWare.Framework/Model/Profiles/MMessageProfile.cs
using GrowthWare.Framework.Model.Profiles.Base;
using GrowthWare.Framework.Model.Profiles.Interfaces;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Model.Profiles
{
	/// <summary>
	/// Class MMessageProfile
	/// </summary>
    [Serializable(), CLSCompliant(true)]
    public class MMessageProfile : MProfile, IMessageProfile
    {
        #region "Member Properties"
        private int m_SE_SEQ_ID = 1;
        private string m_Description = string.Empty;
        private string m_Title = string.Empty;
        private bool m_FormatAsHTML = false;
        private string m_Body = string.Empty;
        #endregion

        #region "Private Methods"
        /// <summary>
        /// Initializes values given a DataRow
        /// </summary>
        /// <param name="datarow">datarow</param>
        /// <remarks>Does not set ID or Name .. ColumnName should be unique to
        /// each inheriting class.</remarks>
        protected new void Initialize(DataRow datarow)
        {
            base.NameColumnName = "Name";
            base.IdColumnName = "Message_SeqID";
            base.Initialize(datarow);
            m_SE_SEQ_ID = base.GetInt(datarow, "SE_SEQ_ID");
            m_Title = base.GetString( datarow, "TITLE");
            m_Description = base.GetString(datarow, "DESCRIPTION");
            m_FormatAsHTML = base.GetBool(datarow, "FORMAT_AS_HTML");
            m_Body = base.GetString(datarow, "BODY");
        }
        #endregion

        #region "Public Methods"
        /// <summary>
        /// Will return a message profile with the default values
        /// </summary>
        /// <remarks></remarks>
        public MMessageProfile()
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="MMessageProfile" /> class.
		/// </summary>
		/// <param name="profile">MMessageProfile</param>
        public MMessageProfile(MMessageProfile profile)
		{
			if (profile != null) 
			{
				this.AddedBy = profile.AddedBy;
				this.AddedDate = profile.AddedDate;
				this.Body = profile.Body;
				this.Description = profile.Description;
				this.FormatAsHTML = profile.FormatAsHTML;
				this.Id = profile.Id;
				this.Name = profile.Name;
				this.SE_SEQ_ID = profile.SE_SEQ_ID;
				this.Title = profile.Title;
				this.UpdatedBy = profile.UpdatedBy;
				this.UpdatedDate = profile.UpdatedDate;
			}
		}

        /// <summary>
        /// Initializes a new instance of the <see cref="MMessageProfile" /> class.
        /// </summary>
        /// <param name="dr">The dr.</param>
        public MMessageProfile(DataRow dr)
        {
            this.Initialize(dr);
        }

        /// <summary>
        /// Formats the body.
        /// </summary>
        void IMessageProfile.FormatBody()
        {
            PropertyInfo[] myPropertyInfo = this.GetType().GetProperties();
            PropertyInfo myPropertyItem = null;
            foreach (PropertyInfo myPropertyItem_loopVariable in myPropertyInfo)
            {
                myPropertyItem = myPropertyItem_loopVariable;
                object pValue = myPropertyItem.GetValue(this, null);
                if (pValue != null)
                {
                    this.m_Body = this.m_Body.Replace("<" + myPropertyItem.Name + ">", pValue.ToString());
                }
            }
        }

        /// <summary>
        /// Returns all properties encapsulated by angle brackets seporated by the Seporator parameter
        /// </summary>
        /// <param name="seporator">string</param>
        /// <returns>string</returns>
        public string GetTags(string seporator)
        {
            string retVal = string.Empty;
            PropertyInfo[] mPropertyInfo = this.GetType().GetProperties();
            foreach (PropertyInfo mPropertyItem in mPropertyInfo)
            {
                retVal = retVal + "<" + mPropertyItem.Name + ">" + seporator;
            }
            return retVal;
        }
        #endregion

        #region "Public Properties"
        /// <summary>
        /// Gets or sets the S e_ SE q_ ID.
        /// </summary>
        /// <value>The Security Entity ID.</value>
        public int SE_SEQ_ID
        {
            get { return m_SE_SEQ_ID; }
            set { m_SE_SEQ_ID = value; }
        }

        /// <summary>
        /// Gets or sets the title.
        /// </summary>
        /// <value>The title.</value>
        public string Title
        {
            get { return m_Title; }
            set { m_Title = value.Trim(); }
        }

        /// <summary>
        /// Gets or sets the description.
        /// </summary>
        /// <value>The description.</value>
        public string Description
        {
            get { return m_Description; }
            set { m_Description = value.Trim(); }
        }

        /// <summary>
        /// Gets or sets a value indicating whether [format as HTML].
        /// </summary>
        /// <value><c>true</c> if [format as HTML]; otherwise, <c>false</c>.</value>
        public bool FormatAsHTML
        {
            get { return m_FormatAsHTML; }
            set { m_FormatAsHTML = value; }
        }

        /// <summary>
        /// Gets or sets the body.
        /// </summary>
        /// <value>The body.</value>
        public string Body
        {
            get { return m_Body; }
            set { m_Body = value.Trim(); }
        }
        #endregion
    }
}
